lib/suse-cloud-upgrade-4-to-5-pre

```diff
@@ -38,6 +39,37 @@ fi
 echo_summary "Performing sanity checks..."
 
 
+### Check if repositories are prepared for the upgrade
+# We need to ensure that Cloud5 main repository and update repos are in place
+
+# Skip the repository checks if they are defined in provisioner.json:
+# there might be non-default (and/or network) path
+if [ -f /etc/crowbar/provisioner.json ]; then
+  PROVISIONER_JSON=/etc/crowbar/provisioner.json
+  for repo in SLE-Cloud SUSE-Cloud-5-Pool SUSE-Cloud-5-Updates; do
```

Comment:
Can we move the code after https://github.com/jsuchome/suse-cloud-upgrade/blob/repo_checks/lib/suse-cloud-upgrade-4-to-5-pre#L149 ? I think it's better to first have the check that provisioner.json is correct.

Also, I realized that this checks the SMT mirror, but not the repos registered on the admin server. I'm not sure how we can best do that (because we really want that -- it's important to use the updates for barclamps). Any idea?